### DISCUSSION TITLE
Schema Relation Structuring Question

### DISCUSSION THREAD

**Discussion by Rykuno** [prisma/prisma#10401] [Q&A]
Say you have two entities, `user` and `community`. Each have their own unique fields but share relations like `Post`.

I'm trying to figure out if there is a right way to structure this. 

(1) Access `Post` through `User`. I can check to see if user has permission to view the post before fetching based.

- (BAD) I'd need separate GQL queries for both `User` and `Community` and updating the cache gets more complex needing both a  `useCreateUserPost` and `useCreateCommunityPost`.
- (GOOD) My permission checks on the backend are easier. 

```gql
query user($id: String!, $postId: String!) {
  user(where: { id: { equals: $id } }) {
    id
    post(where: { id: { equals: $postId } }) {
      id
      text
    }
  }
}
```

(2) I can directly access the post. I can check after the post is fetched to see if the user has permission to view.

- (BAD) On the API i'd have to fetch the post, check if its a `user` or `community` post, then run permission checks
- (GOOD) Can use a single hook `useCreatePost` or `useFetchPost` for both `user` and `community` pages.

```gql
query post($id: String!) {
  post(where: { id: { equals: $id } }) {
    id
    text
  }
}
```

I suppose when is it a good idea to do one or the other in cases like this?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
